### ISSUE TITLE
battle box scaling needs improvement

### ISSUE THREAD

**Issue by ventus96** [ventus96/soultest#3]
The function scalebb doesn't quite scale the battle box to the dimensions that are put into it. It gets close to what it is but then needs to have it set at the end of the function. This makes the dimensions jump suddenly after they reach the end point.
(Also it may have to be linear to represent Undertale's scaling properly?)

The problem is that to avoid subpixeling it was implemented with it being drawn with math.floor (and the width and height given that extra 0.5 pixel) each time so it can get stuck reaching the end point if the formula doesn't produce a number large enough to add a whole pixel.

(If it did use linear scaling the rounding would still be a problem as it would make the result different at the end of the function due to rounding)

tldr; scalebb function needs to be made more smooth.
